### PR TITLE
Add analytics

### PR DESCRIPTION
**Pull request by Deren-Liao**
* Add analytics for CSR feature.
* Renamed const from DeploymentDurationProperty  to DurationProperty for some classes to be more appropriate.

### CODE REVIEW

**Comment by Deren-Liao** [comment]
Ping,  could you please take a look?  The other 3 PRs too.   
With these, I am ready to merge CSR feature to master.  

Thanks.

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrConnectSectionOpenEvent.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace GoogleCloudExtension.Analytics.Events
+{
+    internal static class CsrConnectSectionOpenEvent
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrConnectSectionOpenEvent.cs]: Class comment?

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryClonedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// limitations under the License.
+
+using System;
+
+namespace GoogleCloudExtension.Analytics.Events
+{
+    /// <summary>
+    /// This event is sent after a CSR repo is cloned.
+    /// </summary>
+    internal static class CsrRepositoryClonedEvent
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryClonedEvent.cs]: nit: isnt the R in CSR repository?  So maybe "CsrClonedEvent"?

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryCreatedEvent.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// limitations under the License.
+
+using System;
+
+namespace GoogleCloudExtension.Analytics.Events
+{
+    /// <summary>
+    /// This event is sent after a CSR repo is created.
+    /// </summary>
+    internal static class CsrRepositoryCreatedEvent
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryCreatedEvent.cs]: Same here

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryListedEvent.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// limitations under the License.
+
+using System;
+
+namespace GoogleCloudExtension.Analytics.Events
+{
+    /// <summary>
+    /// This event is sent after CSR repos are shown in Team Explorer.
+    /// </summary>
+    internal static class CsrRepositoryListedEvent
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryListedEvent.cs]: and here

Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryListedEvent.cs]: Done

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryCreatedEvent.cs (hunk 3 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryCreatedEvent.cs]: Done

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryClonedEvent.cs (hunk 2 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrRepositoryClonedEvent.cs]: Done

File: GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrConnectSectionOpenEvent.cs (hunk 1 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/Analytics/Events/CsrConnectSectionOpenEvent.cs]: Done

**Comment by Deren-Liao** [comment]
Renamed and added comment as Ian commented.  Thanks for review.

**Pull request merged**